contracts/vortex/CarbonVortex.sol

```diff
@@ -252,37 +858,109 @@ contract CarbonVortex is ICarbonVortex, Upgradeable, ReentrancyGuardUpgradeable,
         emit RewardsUpdated({ prevRewardsPPM: prevRewardsPPM, newRewardsPPM: newRewardsPPM });
     }
 
+    function _setPairDisabled(Token token) private {
+        bool prevPairStatus = _disabledPairs[token];
+
+        _disabledPairs[token] = !prevPairStatus;
+
+        emit PairDisabledStatusUpdated(token, prevPairStatus, !prevPairStatus);
+    }
+
     /**
-     * @dev set allowance to Bancor Network V3 to the max amount if it's less than the input amount
+     * @dev returns true if the auction price is below or equal to the minimum possible price
+     * @dev check if timeElapsed / priceDecayHalfLife >= 128
      */
-    function _setAllowance(Token token, uint256 inputAmount) private {
-        if (token.isNative()) {
-            return;
+    function _auctionPriceIsBelowMinimum(Token token) private view returns (bool) {
+        // cache trading start time to save gas
+        uint32 tradingStartTime = _tradingStartTimes[token];
+        // trading hasn't been enabled, return false
+        if (tradingStartTime == 0) {
+            return false;
         }
-        uint256 allowance = token.allowance(address(this), address(_bancorNetwork));
-        if (allowance < inputAmount) {
-            // increase allowance to the max amount if allowance < inputAmount
-            token.safeIncreaseAllowance(address(_bancorNetwork), type(uint256).max - allowance);
+        // get time elapsed since trading was enabled
+        uint32 timeElapsed = uint32(block.timestamp) - tradingStartTime;
+        // get the halflife for the token
+        uint32 currentPriceDecayHalfLife = token == _targetToken ? _targetTokenPriceDecayHalfLife : _priceDecayHalfLife;
+        // check if the maximum amount of halflifes have been reached
+        if (timeElapsed / currentPriceDecayHalfLife >= 128) {
+            return true;
+        } else {
+            return false;
         }
```

Comment:
no need to continue the auction at this stage, since price is 0, even without reverting the exp decay calculation.

Reply:
So what exactly are you checking `timeElapsed / currentPriceDecayHalfLife >= 128` for?

I can say that it looks compatible with the restriction that I've **previously** mentioned in the ExpDecay library documentation. But I've updated that restriction since, and I suspect that you might be relying on the previous definition of it.